src/main/kotlin/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtSharedTableBackupManager.kt

```diff
@@ -0,0 +1,456 @@
+            val rowsToInsert: List<TenantTableRow> = gson.fromJson(backupFile.objectContent.bufferedReader(),
+                    object : TypeToken<List<TenantTableRow>>() {}.type)
+            for (row in rowsToInsert) {
+                mtContext.withContext(restoreMtBackupRequest.newTenantTable.tenantName) {
+                    sharedTableMtDynamo.putItem(restoreMtBackupRequest.newTenantTable.virtualTableName, row.attributeMap)
+                }
+            }
+        }
+        logger.info("${restoreMtBackupRequest.backupName}: Finished restoring ${restoreMtBackupRequest.tenantTableBackup}" +
+                " to ${restoreMtBackupRequest.newTenantTable} in ${System.currentTimeMillis() - startTime} ms")
+
+        return TenantRestoreMetadata(restoreMtBackupRequest.backupName,
+                Status.COMPLETE,
+                restoreMtBackupRequest.newTenantTable.tenantName,
+                restoreMtBackupRequest.newTenantTable.virtualTableName)
+    }
+
+    private fun getTenantTableMetadata(backupId: String, tenantTable: TenantTable): CreateTableRequest {
+        val tenantTableMetadataS3Location = getTenantTableMetadataFile(backupId, tenantTable)
+        val tenantTableMetadataFile: S3Object = s3.getObject(s3BucketName, tenantTableMetadataS3Location)
+        return gson.fromJson<CreateTableRequest>(tenantTableMetadataFile.objectContent.bufferedReader(),
+                CreateTableRequest::class.java)
+    }
+
+    private fun getBackupFileKeys(backupId: String, tenantTable: TenantTable): Set<String> {
+        val ret = Sets.newHashSet<String>()
+        var continuationToken: String? = null
+        do {
+            val listBucketResult: ListObjectsV2Result = s3.listObjectsV2(
+                    ListObjectsV2Request()
+                            .withBucketName(s3BucketName)
+                            .withContinuationToken(continuationToken)
+                            .withPrefix("$backupDir/$backupId/${tenantTable.tenantName}/${tenantTable.virtualTableName}"))
+            continuationToken = listBucketResult.continuationToken
+            for (o in listBucketResult.objectSummaries) {
+                ret.add(o.key)
+            }
+        } while (listBucketResult.isTruncated)
+        return ret
+    }
+
+    override fun listBackups(listBackupRequest: ListBackupsRequest): ListBackupsResult {
+        val ret = Lists.newArrayList<BackupSummary>()
+
+        Preconditions.checkArgument(listBackupRequest.backupType == null, "Listing backups by backupType unsupported")
+        Preconditions.checkArgument(listBackupRequest.tableName == null, "Listing backups by table name unsupported for multi tenant backups")
+        Preconditions.checkArgument(listBackupRequest.timeRangeLowerBound == null, "Listing backups filtered by time range unsupported [currently]")
+        Preconditions.checkArgument(listBackupRequest.timeRangeUpperBound == null, "Listing backups filtered by time range unsupported [currently]")
```

Comment:
It would be good to start thinking about how we want to lay out backup metadata in DynamoDB.

Reply:
sure, will keep that close to the top of the next steps.